### PR TITLE
[data/preprocessors] do not fail transform_batch on missing column

### LINKED ISSUE #48047: [Data]: SimpleImputer fails on transform_batch when column is completely missing from dataframe
### What happened + What you expected to happen

When providing a pandas dataframe without the expected column to impute a missing value for, the Imputer fails with a `KeyError` on this [line](https://github.com/ray-project/ray/blob/master/python/ray/data/preprocessors/imputer.py#L131).

The expected behavior would be that the column is added and imputed if it is missing from the dataframe.

### Versions / Dependencies

Latest ray

### Reproduction script

```python
import ray
from ray.data.preprocessors import SimpleImputer
import pandas as pd
import nump as np

ds = ray.data.from_items([{"A": 1, "B", np.nan}])
imputer = SimpleImputer(columns=list("B"), strategy="constant", fill_value=0)

fitted_imputer = imputer.fit(ds)

fitted_imputer.transform_batch(pd.DataFrame({"A": [1], "B": [np.nan]}))
#     A    B
# 0  1  0.0

fitted_imputer.transform_batch(pd.DataFrame({"A": [1]}))
# fails with `KeyError: B`
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by martinbomio**
## Why are these changes needed?

[](https://github.com/ray-project/ray/issues/48047)

## Related issue number

Closes https://github.com/ray-project/ray/issues/48047

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/tests/preprocessors/test_imputer.py [hunk 1]
```diff
@@ -113,6 +125,8 @@ def test_simple_imputer():
 
     assert constant_out_df.equals(constant_expected_df)
 
+    #
```
Comment (richardliaw) [inline python/ray/data/tests/preprocessors/test_imputer.py:128]: cleanup?

File: python/ray/data/preprocessors/imputer.py [hunk 2]
```diff
@@ -131,7 +131,14 @@ def _transform_pandas(self, df: pd.DataFrame):
                 if is_categorical_dtype(df.dtypes[column]):
                     df[column] = df[column].cat.add_categories(value)
 
-        df = df.fillna(new_values)
+        for column_name in new_values:
```
Comment (richardliaw) [inline python/ray/data/preprocessors/imputer.py:134]: can you add this behavior to the docstring as well?

**Review by richardliaw — APPROVED** [review]

File: python/ray/data/preprocessors/imputer.py (hunk 2 above)
Comment (martinbomio) [reply python/ray/data/preprocessors/imputer.py:135]: done!

File: python/ray/data/preprocessors/imputer.py [hunk 3]
```diff
@@ -13,7 +13,8 @@
 
 @PublicAPI(stability="alpha")
 class SimpleImputer(Preprocessor):
-    """Replace missing values with imputed values.
+    """Replace missing values with imputed values. If the column is missing from a
+    batch, it will be filled with the imputed valuegit.
```
Comment (omatthew98) [inline python/ray/data/preprocessors/imputer.py:17]: Nit: `valuegit` to `value`

**Review by omatthew98 — APPROVED** [review]

**Comment by martinbomio** [comment]
@omatthew98 @richardliaw would you mind merging, I can't

**Review by richardliaw — APPROVED** [review]

**Pull request merged**